### PR TITLE
[#IOCIT-346] Add storage for lollipop assertions

### PR DESCRIPTION
**Pull request by fabriziopapi**
depends on https://github.com/pagopa/terraform-azurerm-v3/pull/67

### List of changes
see commit history

### Motivation and context
Add a storage to save the spid assertions used by lollipop in a blob. 
This storage will be encrypted with a customer managed key.

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
there is something strange here
https://github.com/pagopa/io-infra/blob/IOCIT-346--add-assertions-storage-for-lollipop/src/domains/citizen-auth-common/99_locals.tf#L2

in common dir we have multiregion resources so ${var.location_short} should be removed

**Comment by fabriziopapi** [comment]
> there is something strange here https://github.com/pagopa/io-infra/blob/IOCIT-346--add-assertions-storage-for-lollipop/src/domains/citizen-auth-common/99_locals.tf#L2
> 
> in common dir we have multiregion resources so ${var.location_short} should be removed

you are right, reverted to var.project.
Thanks!

File: src/domains/citizen-auth-common/03_storage.tf [hunk 1]
```diff
@@ -0,0 +1,60 @@
+resource "azurerm_resource_group" "lollipop_assertions_rg" {
+  name     = format("%s-lollipop-common-rg", local.product)
+  location = var.location
+
+  tags = var.tags
+}
+
+module "lollipop_assertions_storage" {
+  source                     = "git::https://github.com/pagopa/terraform-azurerm-v3//storage_account?ref=v4.3.1"
+  name                       = replace(format("%s-assertions", local.product), "-", "")
```
Comment (pasqualedevita) [inline src/domains/citizen-auth-common/03_storage.tf:10]:
```suggestion
  name                       = replace(format("%s-assertions", local.product), "-", "")
```
from plan the name will be `iopassertions`, is it expected? or maybe `ioplollipopassertionsst`?

Comment (fabriziopapi) [reply src/domains/citizen-auth-common/03_storage.tf]: it should be short enough. Changed! 👍

**Pull request merged**